crates/ruma-common/src/events/relation.rs

```diff
@@ -239,6 +239,7 @@ impl BundledThread {
 /// [reference]: https://spec.matrix.org/v1.5/client-server-api/#reference-relations
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[serde(tag = "rel_type", rename = "m.reference")]
```

Comment:
Other types in this module don't have the attribute, why is it needed here and nowhere else?

Reply:
`Annotation` has it too. It's needed when the struct is included directly in the type outside of a `Relation` enum, because the type only accepts a single relation type (ie `Reference` for key verification events or `Annotation` for reactions).

Luckily it doesn't interfere with the `Relation` enums. We could probably set it for all relation structs, it shouldn't hurt.